Assets/Scripts/Agent/NpcMove.cs

```diff
@@ -0,0 +1,233 @@
+    public int frames = 0;
+    public Vector3 targetVector;
+    List<string> animationConditions = new List<string>() {
+      "Talking","Dancing","Idle"
+    };
+    public float speed = 0.5f;
+    Animator animator;
+
+    private bool isGoalSet = false;
+    private bool isGoalAchieved = false;
+    private Interactable[] possibleGoals;
+    private Interactable currentGoal;
+    private NavMeshPath path;
+    private bool pathSet = false;
+    private float elapsed = 0f;
+    // private PickUpable currentHeldItem;
+    //
+    // public Transform pickupDestination;
+
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        navMeshAgent = this.GetComponent<NavMeshAgent>();
+        // animator = this.GetComponent<Animator>();
+        path = new NavMeshPath();
+        current_animation_state = "Idle";
+        // current_animation_state = idle;
+    }
+
+    // possibleGoals = checkEnvironment(); returns interactables in the environment
+    // setGoal(possibleGoals); might be to get to somewhere, or interact with an interactable
+    // in a loop, maybe achieve goal or perhaps with probability do a set of actions in an intemediate stage to doing the goal.
+    // if goal achieved, checkEnvironment() and start again.
+
+    private Interactable[] checkEnvironment() {
+      Interactable[] interactables = GameObject.FindObjectsOfType<Interactable>();
+      return interactables;
+    }
+
+    private Interactable setGoal(Interactable[] possGoals){
+      System.Random r = new System.Random(System.Guid.NewGuid().GetHashCode());
+      int randInteractable = r.Next(possGoals.Length);
+      return (Interactable)possGoals[randInteractable];
+    }
+
+    private bool attemptGoal(Interactable currGoal) {
+      //check how far you are from object
+      //if not interactable radius, then walk towards object, return false
+      //else interact with interactable, return true
+      float distance = getDistance(currGoal);
+      if (distance > 3f) {
+         // walkPath(currGoal);
+         // string possibleAction = get_animation_condition();
+         // doIntermediaryAction(possibleAction);
+         return false;
+      }
+      else {
+        frames++;
+        if(frames == 500){
+          frames = 0;
+          interactWithInteractable(currGoal);
+          Debug.Log("interact");
+          return true;
+        } else {
+          return false;
+        }
+
+        // interactWithInteractable(currGoal);
+        // return true;
+      }
+    }
+
+    // private void doIntermediaryAction(string action){
+    //
+    // }
+
+    private float getDistance(Interactable currGoal){
+      float dist = Vector3.Distance(currGoal.transform.position, transform.position);
+      return dist;
+    }
+
+    // private void walkPath(Interactable currGoal){
+    //
+    // }
+
+    private void interactWithInteractable(Interactable currGoal){
+      // if (currGoal is PickUpable) {
```

Comment:
This can now definitely go as the pickup logic is all handled in the interactable